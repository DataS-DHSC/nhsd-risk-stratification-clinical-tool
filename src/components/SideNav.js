import React from 'react';
import { ContentsList } from 'nhsuk-react-components';
import PropTypes from 'prop-types';

const SideNav = ({ children }) => (
  <div className="article-nav">
    <h2 className="article-nav__title"> Page contents</h2>
    <ContentsList
      className="article-nav-list"
      visuallyHiddenText={false}
      aria-label="Pages in this guide"
    >
      {children}
    </ContentsList>
  </div>
);

const Item = ({ children, ...rest }) => (
  <ContentsList.Item className="article-nav-list__item" {...rest}>
    {children}
  </ContentsList.Item>
);

Item.propTypes = {
  children: PropTypes.node.isRequired,
  rest: PropTypes.shape(),
};

Item.defaultProps = {
  rest: {},
};

SideNav.propTypes = {
  children: PropTypes.node.isRequired,
};

SideNav.Item = Item;

export default SideNav;
