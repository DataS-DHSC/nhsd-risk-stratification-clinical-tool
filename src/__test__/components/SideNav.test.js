import React from 'react';
import { render } from '@testing-library/react';
import SideNav from '../../components/SideNav';

const setup = (children) => {
  const props = {
    children,
  };

  return render(<SideNav {...props} />);
};

describe('SideNav', () => {
  it('should match snapshot with no child elements', () => {
    const container = setup('');
    expect(container.asFragment()).toMatchSnapshot();
  });

  it('should match snapshot with child elements', () => {
    const container = setup(<SideNav.Item href="#">Link</SideNav.Item>);
    expect(container.asFragment()).toMatchSnapshot();
  });
});
