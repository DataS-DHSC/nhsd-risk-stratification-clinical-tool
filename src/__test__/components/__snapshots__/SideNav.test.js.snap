// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav should match snapshot with child elements 1`] = `
<DocumentFragment>
  <div
    class="article-nav"
  >
    <h2
      class="article-nav__title"
    >
       Page contents
    </h2>
    <nav
      aria-label="Pages in this guide"
      class="nhsuk-contents-list article-nav-list"
      role="navigation"
    >
      <ol
        class="nhsuk-contents-list__list"
      >
        <li
          class="nhsuk-contents-list__item article-nav-list__item"
        >
          <a
            class="nhsuk-contents-list__link"
            href="#"
            rest="[object Object]"
          >
            Link
          </a>
        </li>
      </ol>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`SideNav should match snapshot with no child elements 1`] = `
<DocumentFragment>
  <div
    class="article-nav"
  >
    <h2
      class="article-nav__title"
    >
       Page contents
    </h2>
    <nav
      aria-label="Pages in this guide"
      class="nhsuk-contents-list article-nav-list"
      role="navigation"
    >
      <ol
        class="nhsuk-contents-list__list"
      />
    </nav>
  </div>
</DocumentFragment>
`;
